synapse/handlers/room_member.py

```diff
@@ -272,6 +272,7 @@ async def _local_membership_update(
         allow_no_prev_events: bool = False,
         prev_event_ids: Optional[List[str]] = None,
         auth_event_ids: Optional[List[str]] = None,
```

Comment:
Since we are now using the new `state_event_ids` parameter to do all of the heavy lifting. I think we can remove any spots where we plumbed `auth_event_ids` just for MSC2716 things in https://github.com/matrix-org/synapse/pull/9247/files. This includes the following functions:

 - `create_and_send_nonmember_event`
 - `_local_membership_update`
 - `update_membership`
 - `update_membership_locked`

We do need to keep `prev_event_ids` as that's still important and separate from the state/auth.


Should I do this as part of this PR or is it okay to split that out to another PR as a fast follow-up?

Reply:
I think it's fine to keep it separate.

Reply:
Created https://github.com/matrix-org/synapse/pull/12304 to take care of this ⏩